### PR TITLE
add conversion scripts for EUPE

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

As per title. Adds the 3 convnext variants and 3 ViT variants of EUPE.
Fixes #45322

### CODE REVIEW

File: src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py [hunk 1]
```diff
@@ -77,6 +83,8 @@ def get_dinov3_config(model_name: str) -> DINOv3ConvNextConfig:
             depths=[3, 3, 27, 3],
             hidden_sizes=[192, 384, 768, 1536],
         )
+    elif model_name in ("eupe_convnext_tiny", "eupe_convnext_small", "eupe_convnext_base"):
```
Comment (vasqu) [inline src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py]: Nit: we can avoid the recursive call tbh and just remove the prefix in any case, should be a no op for existing ones and fixes this one, no?

File: src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py [hunk 2]
```diff
@@ -167,35 +178,36 @@ def convert_and_test_dinov3_checkpoint(args):
     torch.testing.assert_close(original_pixel_values, inputs["pixel_values"], atol=1e-6, rtol=1e-6)
     print("Preprocessing looks ok!")
 
-    with torch.inference_mode(), torch.autocast("cuda", dtype=torch.float):
-        model_output = model(**inputs)
+    if f"{model_name}_cls" in expected_outputs:
```
Comment (vasqu) [inline src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py]: Shouldnt we rather add values then?

**Review by vasqu — APPROVED** [review]
Just some smaller comments but overall looks good 🫡 neat

**Comment by molbap** [comment]
Thanks for the review! I'm a bit against adding slices to the convert script because it's not really a test contract (integration tests solely for EUPE would do better for instance?) But since I had the logits for original impl, added it

File: src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py (hunk 1 above)
Comment (molbap) [reply src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py]: ah indeed

Comment (molbap) [reply src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py]: fixed!

File: src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py (hunk 2 above)
Comment (molbap) [reply src/transformers/models/dinov3_convnext/convert_dinov3_convnext_to_hf.py]: done :)
